src/main/java/org/primefaces/extensions/component/timeago/TimeAgo.java

```diff
@@ -0,0 +1,185 @@
+import org.primefaces.component.api.Widget;
+import org.primefaces.extensions.util.MessageFactory;
+import org.primefaces.util.ComponentUtils;
+import org.primefaces.util.ResourceUtils;
+
+/**
+ * <code>TimeAgo</code> component.
+ *
+ * @author Jasper de Vries &lt;jepsar@gmail.com&gt;
+ * @since 7.0.1
+ */
+@ResourceDependencies({
+    @ResourceDependency(library = "primefaces", name = "jquery/jquery.js"),
+    @ResourceDependency(library = "primefaces", name = "core.js"),
+    @ResourceDependency(library = "primefaces-extensions", name = "timeago/jquery.timeago.js"),
+    @ResourceDependency(library = "primefaces-extensions", name = "timeago/timeago-widget.js")
+})
+public class TimeAgo extends UIComponentBase implements Widget {
+
+    public static final String COMPONENT_TYPE = "org.primefaces.extensions.component.TimeAgo";
+    public static final String COMPONENT_FAMILY = "org.primefaces.extensions.component";
+    public static final String DEFAULT_RENDERER = "org.primefaces.extensions.component.TimeAgoRenderer";
+
+    public static final String STYLE_CLASS = "ui-timeago ui-widget";
+
+    private static final List<String> BUNDLED_LOCALES = Arrays.asList(
+            "af",
+            "am",
+            "ar",
+            "az",
+            "be",
+            "bg",
+            "bs",
+            "ca",
+            "cs",
+            "cy",
+            "da",
+            "de",
+            "dv",
+            "el",
+            "en",
+            "es",
+            "et",
+            "eu",
+            "fa",
+            "fi",
+            "fr",
+            "gl",
+            "he",
+            "hr",
+            "hu",
+            "hy",
+            "id",
+            "is",
+            "it",
+            "ja",
+            "jv",
+            "ko",
+            "ky",
+            "lt",
+            "lv",
+            "mk",
+            "nl",
+            "no",
+            "pl",
+            "pt-br",
+            "pt",
+            "ro",
+            "rs",
+            "ru",
+            "rw",
+            "si",
+            "sk",
+            "sl",
+            "sq",
+            "sr",
+            "sv",
+            "th",
+            "tr",
+            "uk",
+            "ur",
+            "uz",
+            "vi",
+            "zh-cn",
+            "zh-tw");
+
+    // @formatter:off
+    public enum PropertyKeys {
+        value,
+        widgetVar,
+        style,
+        styleClass
+    }
+    // @formatter:on
+
+    public TimeAgo() {
+        setRendererType(DEFAULT_RENDERER);
+
+        final String i18nAbbr = getBundledLocaleAbbr(MessageFactory.getLocale());
+        if (i18nAbbr != null) {
+            ResourceUtils.addComponentResource(FacesContext.getCurrentInstance(),
+                                               "timeago/i18n/jquery.timeago." + i18nAbbr + ".js",
+                                               "primefaces-extensions",
+                                               "body");
```

Comment:
Is this okay? I would have preferred this to be last in the head, but I could not get it there (it ended up being loaded too soon).

Reply:
Instead of adding the JS here in the java code just pass the calculated locale to the Widget as a property.  Then in the Widget init() method you can check if locale != null then load the script in the widget like...

```javascript
if (locale) {
   var languageJsURI = PrimeFaces.resources.getFacesResource('/timeago/i18n/jquery.timeago.' + locale+'.js',
                    PrimeFacesExt.RESOURCE_LIBRARY, PrimeFacesExt.VERSION);
    // load script
    PrimeFacesExt.getScript(languageJsURI , $.proxy(function(data, textStatus) {
                    this.timeAgo = $(this.selector).timeago();
                }, this));
} else {
     this.timeAgo = $(this.selector).timeago();
}

```